src/controllers/expenseController.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable eqeqeq */
+'use strict';
+
+const expenseModel = require('../models/expenseModel');
+const userModel = require('../models/userModel');
+
+function createExpense(req, res) {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!spentAt || !title || !amount || !category || !note || !userId) {
+    return res.status(400).send('All fields are required');
+  }
+
+  const user = userModel.getUserById(userId);
+
+  if (!user) {
+    return res.status(400).send('Bad Request');
+  }
+
+  const newExpense = expenseModel.createExpense(
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  );
+
+  res.status(201).json(newExpense);
+}
+
+function getAllExpenses(req, res) {
+  const expenses = expenseModel.getAllExpenses(req.query);
+
+  res.json(expenses);
+}
+
+function getExpenseById(req, res) {
+  const expense = expenseModel.getExpenseById(parseInt(req.params.id));
```

Comment:
This function is unnecessary, as it does the same as the previous one, _getAllExpenses_, when passing the right query.
Meaning, the following code replaces this function:
```
const [expense] = expenseModel.getAllExpenses({ id: parseInt(req.params.id) });
```

Reply:
nice!